packages/router/router.ts

```diff
@@ -862,35 +862,15 @@ export function createRouter(init: RouterInit): Router {
       }
     );
 
-    if (state.initialized) {
-      return router;
-    }
-
-    let lazyMatches = state.matches.filter((m) => m.route.lazy);
-
-    if (lazyMatches.length === 0) {
-      // Kick off initial data load if needed.  Use Pop to avoid modifying history
+    // Kick off initial data load if needed.  Use Pop to avoid modifying history
+    // Note we don't do any handling of lazy here.  For SPA's it'll get handled
+    // in the normal navigation flow.  For SSR it's expected that lazy modules are
+    // resolved prior to router creation since we can't go into a fallbackElement
+    // UI for SSR'd apps
+    if (!state.initialized) {
```

Comment:
This goes back to exactly what we did prior to lazy.  There's no "I'm half iniitalized" concept.  Either _both_ `lazy` _and_ `data` is resolved for initial matches (SSR), or neither is (SPA).